### DISCUSSION TITLE
How does extPay know which user information to get, when we run extpay.getUser()

### DISCUSSION THREAD

**Discussion by Shashank003** [Glench/ExtPay#102] [Q&A] [answered]
How exactly does this function know, which user's information to get from the server.

**Comment by burkybang** [comment, marked as answer]
When the user signs in, your extension sends a request to extensionpay.com, which generates a unique ID and returns it. Your extension saves the ID in Chrome sync storage. When you run extpay.getUser(), it sends the ID to extensionpay.com asking for the user's information.

**Reply by Shashank003** [reply]
I want to map each user to a specific email id, so if there are 2 users( 2 seperate emails) on the same device, he will have to pay for 2 subscription ID's. My chrome extension takes the user email id he is using currently to do his work, while he is using the chrome extension. Will this code work, as a way to implement verification for 2 or more users, but sharing a device, since extpay does not have an option for seat based pricing.

extpay.getUser().then(user => {
if (user.paid && user.email == currentUserEmail) {
// ...
} else {
// ...
}
})

**Reply by JordanTreDaniel** [reply]
I would like to just be able to pass an email into the getUser method. I think that would eliminate concurrency issues and what not. ( I feel like the chrome storage value for the token may get outdated/compromised? Like what happens if you've already had your token made and you don't send it (bc it got cleared)? I guess a login, but I would also like to do that without a window, by passing in an email, plus maybe some security info.)
